### PR TITLE
Add multiple test site support

### LINKED ISSUE #574: System tests: allow more than one `test_site`
**Tell us about your environment**

* **MarkBind Version:** 1.16.0

Currently, we only have a `test/test_site`, which is a single MarkBind website, that we use to do all our system testing.

However, there may be cases where we need more than one MarkBind websites for testing.

For example, in the upcoming tags feature, we want to (1) test that tags in `site.json` overrides tags in pages, and (2) test that tags in individual pages actually work as expected when no overriding occurs. We currently cannot do that as we are [forced to choose one over another](https://github.com/MarkBind/markbind/pull/467#discussion_r246788142), because you cannot test both of them at the same time in one single website.

### PR DESCRIPTION
**Pull request by jamos-tay**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] New feature

Fixes #574 

**What is the rationale for this request?**

Some features, such as tagging, must be tested on different sites since the test cases are mutually exclusive.

**What changes did you make? (Give an overview)**

Allow support for any number of test sites.

Revamped test directory structure:

- Old:
```
test/
  unit/
    (unit tests)
  test_site/
    (site files)
    test.sh
    test.bat
    testUtil/
      (test utils)
```

- New:
```
test/
  unit/
    (unit tests)
  functional/
    test.sh
    test.bat
    testUtil/
      (test utils)
    test_site/
      (test site files)
    test_site_2/
      (test site 2 files)
    test_site etc... /
      (...)
```

Creating a new test site:

- Create a folder in `test/functional`
- Run `markbind init` in the folder, write test site as per normal
- Create an `expected` directory within the folder and populate with expected files
- Add the folder name to these two lines:

test.sh
```
set sites=test_site test_site_2   // Add the folder name here
```

test.bat
```
declare -a sites=("test_site" "test_site_2")   // Add the folder name here
```

Note: Ensure `expected` directory is ignored (Add `expected/*` to the site.json ignore), and that no `src` or `glob` matches any of the expected files (the default glob `**/*.+(md|mbd)` causes some problems)

**Is there anything you'd like reviewers to focus on?**

The current `test_site_2` folder is just there to show the feature works, it stores a fresh init site. Will remove before merge. No edits made to test_site, except for removing 3 lines from the site.json

How do we want to organize the tests? For example, say we have:
- Feature A: Test case A1 and Test case A2 are mutually exclusive
- Feature B: Test case B1 and Test case B2 are mutually exclusive

We could do something like this:
- Site 1: Other tests
- Site 2: A1, B1
- Site 3: A2, B2

### CODE REVIEW

**Comment by jamos-tay** [comment]
Rebased, updated

I placed all the `test_site` related changes in a separate commit, so you should just have to look at the `Add multiple test site functionality` commit

File: test/functional/test.bat [hunk 1]
```diff
@@ -0,0 +1,21 @@
+@ECHO off
+
+set sites=test_site test_site_2
+
+for %%a in (%sites%) do ( 
+	
+	echo(
```
Comment (yamgent) [inline test/functional/test.bat]: Should be 4 spaces, not `\t`.

File: test/functional/test.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+declare -a sites=("test_site" "test_site_2")
+
+for site in "${sites[@]}"
+do
+	# print site name
+	echo
```
Comment (yamgent) [inline test/functional/test.sh]: Same here with the tabs, should be 4 spaces.

File: test/functional/testExternalScripts.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+The external script [MathJax 2.75](https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML) should be included, and the following MathJax content should render:
```
Comment (yamgent) [inline test/functional/testExternalScripts.md]: These files shouldn't be outside? They are supposed to be inside `test_site`.

EDIT: `test_site` already has these files, so delete this one.

File: test/functional/testLayoutsOverride.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<frontmatter>
```
Comment (yamgent) [inline test/functional/testLayoutsOverride.md]: Same as above, should be inside `test_site`.

File: test/functional/testTagDivs.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Div with shown tag
```
Comment (yamgent) [inline test/functional/testTagDivs.md]: Same as above, should be inside `test_site`.

File: test/functional/testTags.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<frontmatter>
```
Comment (yamgent) [inline test/functional/testTags.md]: Same as above, should be inside `test_site`.

File: test/functional/testExternalScripts.md (hunk 3 above)
Comment (jamos-tay) [reply test/functional/testExternalScripts.md]: My bad I think something went wrong during the rebase process

**Comment by jamos-tay** [comment]
Rebased, updated

**Comment by jamos-tay** [comment]
Yup, removed it

**Comment by jamos-tay** [comment]
Rebased to fix conflicts

**Pull request merged**